src/misc.jl

```diff
@@ -75,6 +95,13 @@ function indexmap{T}(a::AbstractArray{T})
     return d
 end
 
+
+"""
+    levelsmap(a) -> Dict
+
+Construct a dictionary that associates the unique elements of
+`a` with the integers 1 to `length(a)`.
```

Comment:
Looks incorrect to me. Isn't that "with their rank of appearance"? (I must say I don't understand the purpose of this function: this can easily be computed from `unique`... The package could use some cleanup.)